app/utils/aggregated.py

```diff
@@ -255,6 +255,35 @@ def get_latest_cli_version() -> Tuple[Version, str]:
         return Version('0.0.0'), ''
 
 
+def get_version_compatibility(version: str):
+    """
+    Get the version compatibility matrix from the backend.
+    Args:
+        version (str): The version to check.
+    Returns:
+        no return, will raise exception if the version is not compatible.
+    """
+    result = {}
+    try:
+        httpx_client = BaseClient(AppConfig.Connections.url_bff)
+        response = httpx_client._get('v1/validate/cli/version', params={'version': version})
+        result = response.json().get('result', {})
+    except Exception:
```

Comment:
How about put this exception below the code of result validation below?

Reply:
oh this is a bit hard. since if server finds a invalid version or version mismatch it will return 403 or 404, then `httpx_client._get` will immediately raise error. So it is better except only with http request, and have a separated error handler at `if not reset` statement.

Reply:
Gotcha